### PR TITLE
Simplify unnecessary complex logic

### PR DESCRIPTION
**Pull request by GeertJohan**

### CODE REVIEW

File: h2quic/roundtrip.go [hunk 1]
```diff
@@ -78,21 +78,21 @@ func (r *RoundTripper) RoundTripOpt(req *http.Request, opt RoundTripOpt) (*http.
...
 		return nil, fmt.Errorf("quic: unsupported protocol scheme: %s", req.URL.Scheme)
 	}
+	
+	for k, vv := range req.Header {
+		if !httpguts.ValidHeaderFieldName(k) {
+			return nil, fmt.Errorf("quic: invalid http header field name %q", k)
+		}
+		for _, v := range vv {
+			if !httpguts.ValidHeaderFieldValue(v) {
+				return nil, fmt.Errorf("quic: invalid http header field value %q for key %v", v, k)
```
Comment (GeertJohan) [inline h2quic/roundtrip.go]: Shouldn't these errors be prefixed `h2quic`? QUIC itself doesn't care about HTTP headers. But h2quic does.

Comment (lucas-clemente) [reply h2quic/roundtrip.go]: Yeah, changing it to h2quic makes sense. Thanks!

**Comment by marten-seemann** [comment]
@GeertJohan: Have you given up on this PR?

**Comment by marten-seemann** [comment]
Closing due to age.

**Pull request closed without merging**